mlrun/feature_store/retrieval/job.py

```diff
@@ -107,7 +107,7 @@ def set_default_resources(resources, setter_function):
             "order_by": order_by,
             "engine_args": engine_args,
         },
-        inputs={"entity_rows": entity_rows},
+        inputs={"entity_rows": entity_rows} if entity_rows is not None else {},
```

Comment:
what caused this to be discovered?

Reply:
because we moved the validation of inputs to after the enrichment, we now validate the run object and not only the inputs param in the `run` method (we require "str" type keys and values)